x-pack/plugins/ml/public/embeddables/single_metric_viewer/embeddable_single_metric_viewer_container.tsx

```diff
@@ -0,0 +1,212 @@
+import { useSingleMetricViwerInputResolver } from './use_single_metric_viewer_input_resolver';
+import type { ISingleMetricViewerEmbeddable } from './single_metric_viewer_embeddable';
+import type {
+  SingleMetricViewerEmbeddableInput,
+  AnomalyChartsEmbeddableOutput,
+  SingleMetricViewerEmbeddableServices,
+} from '..';
+import { ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE } from '..';
+import { TimeSeriesExplorerEmbeddableChart } from '../../application/timeseriesexplorer/timeseriesexplorer_embeddable_chart';
+import { APP_STATE_ACTION } from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+import { timeSeriesExplorerProvider } from '../../application/util/timeseriesexplorer_utils';
+import './_index.scss';
+
+const RESIZE_THROTTLE_TIME_MS = 500;
+
+interface AppStateZoom {
+  from?: string;
+  to?: string;
+}
+
+export interface EmbeddableSingleMetricViewerContainerProps {
+  id: string;
+  embeddableContext: InstanceType<ISingleMetricViewerEmbeddable>;
+  embeddableInput: Observable<SingleMetricViewerEmbeddableInput>;
+  services: SingleMetricViewerEmbeddableServices;
+  refresh: Observable<any>;
+  onInputChange: (input: Partial<SingleMetricViewerEmbeddableInput>) => void;
+  onOutputChange: (output: Partial<AnomalyChartsEmbeddableOutput>) => void;
+  onRenderComplete: () => void;
+  onLoading: () => void;
+  onError: (error: Error) => void;
+}
+
+export const EmbeddableSingleMetricViewerContainer: FC<
+  EmbeddableSingleMetricViewerContainerProps
+> = ({
+  id,
+  embeddableContext,
+  embeddableInput,
+  services,
+  refresh,
+  onInputChange,
+  onOutputChange,
+  onRenderComplete,
+  onError,
+  onLoading,
+}) => {
+  useEmbeddableExecutionContext<SingleMetricViewerEmbeddableInput>(
+    services[0].executionContext,
+    embeddableInput,
+    ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE,
+    id
+  );
+  const [chartWidth, setChartWidth] = useState<number>(0);
+  const [zoom, setZoom] = useState<AppStateZoom | undefined>();
+  const [selectedForecastId, setSelectedForecastId] = useState<string | undefined>();
+  const [functionDescription, setFunctionDescription] = useState<string | undefined>();
+  const [detectorIndex, setDetectorIndex] = useState<number>(0);
+  const [selectedJob, setSelectedJob] = useState<MlJob | undefined>();
+  const [autoZoomDuration, setAutoZoomDuration] = useState<number | undefined>();
+
+  const { uiSettings } = services[0];
+  const { mlApiServices, mlResultsService } = services[2];
+  const { data, bounds, lastRefresh } = useSingleMetricViwerInputResolver(
+    embeddableInput,
+    refresh,
+    services[1].data.query.timefilter.timefilter,
+    onRenderComplete
+  );
+  const selectedJobId = data?.jobIds[0];
+  const previousRefresh = usePrevious(lastRefresh ?? 0);
+  const mlTimeSeriesExplorer = useMemo(
+    () => timeSeriesExplorerProvider(uiSettings, mlApiServices, mlResultsService),
+    [uiSettings, mlApiServices, mlResultsService]
+  );
```

Comment:
```suggestion
const mlTimeSeriesExplorer = useTimeSeriesExplorerService();
```

Reply:
Updated in f682daaa0b13b541a66cc5725072a61828502541